conformity/fields/meta.py

```diff
@@ -139,3 +140,39 @@ def introspect(self):
             "description": self.description,
             "requirements": [requirement.introspect() for requirement in self.requirements],
         })
+
+
+@attr.s
+class BooleanValidator(Base):
+    """
+    Uses a boolean callable (probably lambda) passed in to validate the value
+    based on if it returns True (valid) or False (invalid).
+    """
+
+    validator = attr.ib()
+    validator_description = attr.ib(validator=attr.validators.instance_of(six.text_type))
+    error = attr.ib(validator=attr.validators.instance_of(six.text_type))
+    description = attr.ib(default=None)
+
+    def errors(self, value):
+        # Run the validator, but catch any errors and return them as an error
+        # as this is maybe in an All next to a type-checker.
+        try:
+            ok = self.validator(value)
+        except Exception:
+            return [
+                Error("Validator errored (invalid type?)"),
```

Comment:
That was the best I could think of. I didn't want to put the exception string in there in case it was sensitive, perhaps that's better?